e3nn/o3/_tensor_product/_sub.py

```diff
@@ -205,6 +205,72 @@ def __init__(
         super().__init__(irreps_in1, irreps_in2, out, instr, irrep_normalization=irrep_normalization, **kwargs)
 
 
+class FullTensorProductSHWeighted(TensorProduct):
```

Comment:
Change the name to avoid confusion: "FullTensorProduct -> SeparableTensorProduct"? We have this in e3tools: https://github.com/prescient-design/e3tools/blob/main/src/e3tools/nn/_tensor_product.py#L8C7-L8C16